flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -746,7 +797,11 @@ public boolean isObjectReuseEnabled() {
     }
 
     public GlobalJobParameters getGlobalJobParameters() {
```

Comment:
We should solve this differently. `GlobalJobParameters` should be a simple `Map<String, String>`. Maybe we keep those fields in `ExecutionConfig` (not backed by Configuration) and deprecate the corresponding setters and getters and replace them with better long-term API.

Even our examples show how to extend from global job parameters:
https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/windowing/TopSpeedWindowing.java#L83

Reply:
The alternative to @zentol's comment would otherwise be thread context classloader.

Reply:
I agree that we should change this to a simple `Map<String, String>` (and force the map to not required the user CL), or maybe a `Configuration`.
IMO this API was never intuitive in the first place.